Client/Application/AppDelegate.swift

```diff
@@ -303,6 +383,10 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
                 url = item.value
             case "private":
                 isPrivate = NSString(string: item.value ?? "false").boolValue
+            case "deep-link":
```

Comment:
I'm not happy with how we now have three different things under the `firefox://` scheme:

- `firefox://fxa-signin?signin=<token>&someQuery=<data>...`
- `firefox://open-url?deep-link=https%253A...`
- `firefox://anything-here-does-not-matter?deeplink=https%253A...`

This reflects in the code, which is handling the last two cases combined and seems all over the place.

I would like to ask for three things:

1) Introduce a `firefox://deep-link?url=` style url
2) Enforce a `firefox://open-url?url=` style for the public API
3) Refactor this code to have three clear separate blocks to handle `firefox://fxa-signin`, `firefox://open-url` and `firefox://deep-link`.